File: ios/Plugin/MicrophoneTypes.swift
//
//  MicrophoneTypes.swift
//  Plugin
//
//  Created by mozart alkhateeb on 30/05/2021.
//  Copyright © 2021 Max Lynch. All rights reserved.
//

struct AudioRecording {
    let base64String: String?
    let dataUrl: String?
    let path: String?
    let webPath: String?
    let duration: Int
    let format: String?
    let mimeType: String?
    
    func toDictionary() -> Dictionary<String, Any> {
        var result: [String: Any] = [:]
        
        if base64String != nil {
            result["base64String"] = base64String
        }
        
        if dataUrl != nil {
            result["dataUrl"] = dataUrl
        }
        
        if path != nil {
            result["path"] = path
        }
        
        if webPath != nil {
            result["webPath"] = webPath
        }
        
        result["duration"] = duration
        
        if format != nil {
            result["format"] = format
        }
        
        if mimeType != nil {
            result["mimeType"] = mimeType
        }
        
        return result
    }
}

enum StatusMessageTypes: String {
    case microphonePermissionNotGranted = "microphone permission not granted"
    case cannotRecordOnThisPhone = "cannot record on this phone"
    case recordingFailed = "recording failed"
    case noRecordingInProgress = "no recording in progress"
    case failedToFetchRecording = "failed to fetch recording"
    case recordingInProgress = "recording in progress"
    case microphoneIsBusy = "microphone is busy"
    case recordingStared = "recording stared"
}

enum MicrophonePermissionType: String, CaseIterable {
    case microphone
}

File: android/src/main/java/com/mozartec/capacitor/microphone/MicrophoneTypes.kt
package com.mozartec.capacitor.microphone

import com.getcapacitor.JSObject

enum class StatusMessageTypes(val value: String) {
    MicrophonePermissionNotGranted("microphone permission not granted"),
    CannotRecordOnThisPhone("cannot record on this phone"),
    RecordingFailed("recording failed"),
    NoRecordingInProgress("no recording in progress"),
    FailedToFetchRecording("failed to fetch recording"),
    RecordingInProgress("recording in progress"),
    MicrophoneIsBusy("microphone is busy"),
    RecordingStared("recording stared")
}

class Recording(val base64String: String?, val dataUrl: String?, val path: String?, val webPath: String?, val duration: Int, val format: String?, val mimeType: String?) {
    fun toJSObject(): JSObject {
        var result = JSObject()

        if (base64String != null) {
            result.put("base64String", base64String)
        }

        if (dataUrl != null) {
            result.put("dataUrl", dataUrl)
        }

        if (path != null) {
            result.put("path", path)
        }

        if (webPath != null) {
            result.put("webPath", webPath)
        }

        result.put("duration", duration)

        if (format != null) {
            result.put("format", format)
        }

        if (mimeType != null) {
            result.put("mimeType", mimeType)
        }

        return result
    }
}